hathor/verification/block_verifier.py

```diff
@@ -51,7 +53,11 @@ def verify_height(self, block: Block) -> None:
 
     def verify_weight(self, block: Block) -> None:
         """Validate minimum block difficulty."""
-        min_block_weight = self._daa.calculate_block_difficulty(block)
+        memory_storage = SimpleMemoryStorage()
+        dependencies = self._daa.get_block_dependencies(block)
+        memory_storage.add_vertices_from_storage(not_none(block.storage), dependencies)
+
+        min_block_weight = self._daa.calculate_block_difficulty(block, memory_storage)
```

Comment:
This is done "inline" here temporarily. In next PRs it will be moved to outside the verification.